src/js/apps/patients/worklist/worklist_sort.js

```diff
@@ -0,0 +1,140 @@
+import { get, union, map, uniq, compact } from 'underscore';
+import Backbone from 'backbone';
+import Radio from 'backbone.radio';
+
+import { alphaSort } from 'js/utils/sorting';
+import { i18n } from 'js/views/patients/worklist/worklist_views';
+
+function getEntityFieldValue(entity, fieldName, keys) {
+  const patientField = entity.getPatient().getField(fieldName);
+  if (!patientField) return null;
+  return get(patientField.getValue(), keys);
+}
+
+// field_key may include a path for the model_attr.deeply_nested.value
+const SortOption = Backbone.Model.extend({
+  defaults: {
+    direction: 'desc',
+    field_key: 'value',
+    entities: ['actions', 'flows'],
+  },
+  getComparator() {
+    const comparator = this.get('comparator');
+    if (comparator) return comparator.bind(this);
+
+    const fieldName = this.get('field_name');
+    const keys = this.get('field_key').split('.');
+    const direction = this.get('direction');
+
+    return (a, b) => {
+      const aValue = getEntityFieldValue(a.model, fieldName, keys);
+      const bValue = getEntityFieldValue(b.model, fieldName, keys);
+
+      return alphaSort(direction, aValue, bValue);
```

Comment:
To resolve the issue with #879, this should read:

```js
return alphaSort(direction, aValue ?? '', bValue ?? '');
```

But to support `type: numeric` that would expand to something like:

```js
if (type === 'numeric') {
  return numSort(direction, aValue ?? 0, bValue ?? 0);
}

return alphaSort(direction, aValue ?? '', bValue ?? '');
```